dynamo/src/main/java/no/bibsys/db/DynamoDBEventProcessor.java

```diff
@@ -38,37 +41,68 @@ public DynamoDBEventProcessor(CloudsearchClient cloudsearchClient) {
     @Override
     public Void handleRequest(DynamodbEvent dynamodbEvent, Context context) {
 
-        logger.debug("dynamodbEvent, #records={}", dynamodbEvent.getRecords().size());
+        try {
+            logger.debug("dynamodbEvent, #records={}, dynamodbEvent={}", dynamodbEvent.getRecords().size(),
+                    dynamodbEvent.toString());
 
-        List<AmazonSdfDTO> documents = dynamodbEvent.getRecords().stream()
-                .map(this::createAmazonSdfFromTriggerEvent)
-                .collect(Collectors.toCollection(ArrayList::new));
+            List<AmazonSdfDTO> documents = dynamodbEvent.getRecords().stream()
+                    .map(this::createAmazonSdfFromTriggerEvent).collect(Collectors.toCollection(ArrayList::new));
 
-        try {
-            cloudsearchClient.uploadbatch(documents);
-        } catch (IOException e) {
-            logger.error("",e);
+            if (!documents.isEmpty()) {
+                cloudsearchClient.uploadbatch(documents);
+            }
+        } catch (Exception e) {
+            logger.error("", e);
         }
         return null;
     }
 
 
     private AmazonSdfDTO createAmazonSdfFromTriggerEvent(DynamodbStreamRecord dynamodDBStreamRecord) {
-        StreamRecord streamRecord = dynamodDBStreamRecord.getDynamodb();
-
-        AmazonSdfDTO sdf = new AmazonSdfDTO(dynamodDBStreamRecord.getEventName());
-        if (streamRecord.getNewImage().containsKey("id")) {
-            sdf.setId(streamRecord.getNewImage().get("id").getS());
-        }
         try {
-            sdf.setFieldsFromEntity(extractFullEntity(streamRecord.getNewImage()));
-        } catch (IOException e) {
+
+
+            StreamRecord streamRecord = dynamodDBStreamRecord.getDynamodb();
+
+            AmazonSdfDTO sdf = new AmazonSdfDTO(dynamodDBStreamRecord.getEventName());
+            if (streamRecord == null) {
```

Comment:
This logic should probably do something as the getKeys calls throw NPEs, also Objects::isNull